### PR TITLE
Display a question summary and allow user's to re-ask them

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
A new question summary is displayed when the answers are displayed by the interactive prompt. The summary will not be provided if the `--answers` flag is used.

The user will also have the opportunity to go back and change any answers. This can be done by either:
* Re-answering the questions with the previous answers pre-populated, or
* Select which questions they would like to answer

Selecting specific questions causes some odd interactions with the "conditional questions" (e.g. 
changing "merge standard error" from `yes` to `no`). This interaction is handled in two ways:
* If the answer of a parent question changes, then the dependent question is asked if the conditional is met, and
* If the user selected a question that no longer meets its condition, an error is emitted to the user and the question is skipped.

```
# Asking additional conditional questions
$ flight job create simple
...
Merge STDERR into STDOUT Yes
...
Which questions would you like to change? (Press ↑/↓/←/→ arrow to scroll, Space/Ctrl+A|R to select (all|rev) and Enter to finish)
  ⬡ Working directory
  ⬡ Set an output file for STDOUT
‣ ⬢ Merge STDERR into STDOUT
  ⬡ Set the name of your job
  ⬡ Do you want email notifications for this job?
  ⬡ Notification email address.
Merge STDERR into STDOUT (Press ↑/↓ arrow to move and Enter to select)
  Yes
‣ No
Set an output file for STDERR (job-%j.error) foo.error
```

```
# Selecting a question which no longer meets its conditional
$ flight job create simple
...
Merge STDERR into STDOUT (Press ↑/↓ arrow to move and Enter to select)
  Yes
‣ No
Set an output file for STDERR (job-%j.error) foo.error
...
Which questions would you like to change? (Press ↑/↓/←/→ arrow to scroll, Space/Ctrl+A|R to select (all|rev) and Enter to finish)
  ⬡ Working directory
  ⬡ Set an output file for STDOUT
  ⬢ Merge STDERR into STDOUT
‣ ⬢ Set an output file for STDERR
  ⬡ Set the name of your job
  ⬡ Do you want email notifications for this job?
Merge STDERR into STDOUT (Press ↑/↓ arrow to move and Enter to select)
‣ Yes
  No
Skipping the following question as the dependencies are no longer met:
Set an output file for STDERR
...
```

### CODE REVIEW

File: lib/flight_job/commands/create_script.rb [hunk 1]
```diff
@@ -32,42 +32,236 @@
...
-              Please provide the answers with the following flag: #{pastel.yellow '--answers'}
-            ERROR
-          elsif $stdout.tty?
-            prompt_answers
-          else
-            msg = <<~WARN.chomp
-              No answers have been provided! Proceeding with the defaults.
+      MULTI_HELP = "(Press ↑/↓/←/→ arrow to scroll, Space/Ctrl+A|R to select (all|rev) and Enter to finish)"
+      SUMMARY = ERB.new(<<~'TEMPLATE', nil, '-')
+        <%= pastel.bold.underline 'SUMMARY' %>
```
Comment (benarmston) [inline lib/flight_job/commands/create_script.rb:37]: A blank line or two before `SUMMARY` would help set it apart from the previous interaction.

File: lib/flight_job/commands/create_script.rb [hunk 2]
```diff
@@ -32,42 +32,236 @@
...
+            true
+          when 'Selected'
+            puts(pastel.yellow(<<~WARN).chomp) if dependencies?
+              WARN: Some of the questions have dependencies on previous answers.
+              The exact question prompts may differ if the dependencies change.
             WARN
-            $stderr.puts pastel.red(msg)
-            FlightJob.logger.warn msg
-            {}
+            opts = { show_help: :always, echo: false, cycle: true, help: MULTI_HELP }
```
Comment (benarmston) [inline lib/flight_job/commands/create_script.rb:90]: `cycle` should be set to false.  I tried to get to the last question by holding down the down key and I didn't get there :-(

**Review by benarmston — CHANGES_REQUESTED** [review]
This seems to work well.  I've added a couple of comments for you to look at.

Whilst you're addressing those, I'll have a think about how the UI/UX here could be improved.  It's not quite as clear about the different steps as it could be; asking questions, displaying a summary, re-asking questions.  The webapp benefits here from simple animations between those, there might be something that can be done here to make it a bit clearer.

Either way, this is looking good.  I like the handling of the dependent questions.

**Comment by WilliamMcCumstie** [comment]
Rebased on master

File: lib/flight_job/commands/create_script.rb (hunk 2 above)
Comment (WilliamMcCumstie) [reply lib/flight_job/commands/create_script.rb]: https://github.com/openflighthpc/flight-job/pull/9/commits/eb6e5e64f72d760a987b5e270a912ff33238643f

File: lib/flight_job/commands/create_script.rb (hunk 1 above)
Comment (WilliamMcCumstie) [reply lib/flight_job/commands/create_script.rb:37]: https://github.com/openflighthpc/flight-job/pull/9/commits/eb6e5e64f72d760a987b5e270a912ff33238643f

**Comment by benarmston** [comment]
What do you think of the following formattings for the summary screen @WilliamMcCumstie ?

The existing format:

image

---

image


---

image

---

image


---

I'm rather partial to the last.  I don't think either the question text or answers will be so long as to cause an issue on sensible sized displays.

**Comment by benarmston** [comment]
Aside from tweaking the formatting, this seems good to me.

**Comment by WilliamMcCumstie** [comment]
@benarmston I will look at implementing the last format. I am currently working a new feature that alters the format a bit anyway. So I will roll the change into that work. There might need to be a couple of alterations to handle `multiline_text`. 

I don't foresee an issue with the size, everything should be fairly sensible. And if not, it will open in a `pager`.

**Comment by benarmston** [comment]
Rolling it into the other work is fine.  I'll point out the newlines surrounding `SUMMARY` and the newline above `Notes`.

**Pull request merged**